sql/core/src/main/scala/org/apache/spark/sql/execution/Window.scala

```diff
@@ -937,6 +939,18 @@ private[execution] object AggregateProcessor {
       imperatives.toArray,
       trackPartitionSize)
   }
+
+  // Tries to find and return `SizeBasedWindowFunction.n` generated on driver side from given
+  // expressions.  Returns executor side global `SizeBasedWindowFunction.n` if none can be found.
+  def windowPartitionSizeAttribute(expressions: Seq[Expression]): AttributeReference = {
```

Comment:
Ah, that's right! Not quite familiar with windowing code path. Thanks for the pointer!